pkg/rules/service.go

```diff
@@ -0,0 +1,196 @@
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	logger.Info("configure alloy rules app")
+	err = s.configureApp(ctx)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	logger.Info("alloy-rules is configured")
+
+	return nil
+}
+
+func (s *Service) CleanUp(ctx context.Context, cluster clusterv1.Cluster) error {
+	logger := log.FromContext(ctx)
+	logger.Info("deleting alloy-rules")
+
+	configmap := configMap()
+
+	err := s.Client.Delete(ctx, configmap)
+	if err != nil && !apierrors.IsNotFound(err) {
+		logger.Error(err, "failed to delete configmap %s", alloyRulesConfigMapName)
+		return errors.WithStack(err)
+	}
+
+	app := app()
+	err = s.Client.Delete(ctx, app)
+	if err != nil && !apierrors.IsNotFound(err) {
+		logger.Error(err, "failed to delete app %s", alloyRulesAppName)
+		return errors.WithStack(err)
+	}
+
+	logger.Info("deleted alloy-rules")
+	return nil
+}
+
+func configMap() *v1.ConfigMap {
+	return &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      alloyRulesConfigMapName,
+			Namespace: alloyRulesAppNamespace,
+			Labels:    labels.Common,
+		},
+	}
+}
+
+func app() *appv1.App {
+	labels := maps.Clone(labels.Common)
+	labels["app-operator.giantswarm.io/version"] = "0.0.0"
+	return &appv1.App{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      alloyRulesAppName,
+			Namespace: alloyRulesAppNamespace,
+			Labels:    labels,
+		},
+	}
+}
+
+func (s Service) createOrUpdateConfigMap(ctx context.Context, cluster clusterv1.Cluster) error {
+	logger := log.FromContext(ctx)
+	// Get list of tenants
+	var tenants []string
+	tenants, err := tenancy.ListTenants(ctx, s.Client)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	configmap := configMap()
+	_, err = controllerutil.CreateOrUpdate(ctx, s.Client, configmap, func() error {
+		values, err := s.generateAlloyConfig(ctx, tenants)
+		if err != nil {
+			logger.Error(err, "failed to generate %s", alloyRulesConfigMapName)
+			return errors.WithStack(err)
+		}
+
+		data := make(map[string]string)
+		data["values"] = values
+
+		configmap.Data = data
+
+		return nil
+	})
+
+	if err != nil {
+		logger.Error(err, "failed to create or update configmap %s", alloyRulesConfigMapName)
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func (s *Service) generateAlloyConfig(ctx context.Context, tenants []string) (string, error) {
+	data := struct {
+		Tenants []string
+	}{
+		Tenants: tenants,
+	}
+
+	var values bytes.Buffer
+	err := appConfigTemplate.Execute(&values, data)
+	if err != nil {
+		return "", err
+	}
+	return values.String(), nil
+}
+
+func (s Service) configureApp(ctx context.Context) error {
+	logger := log.FromContext(ctx)
+
+	app := app()
+	_, err := controllerutil.CreateOrUpdate(ctx, s.Client, app, func() error {
+		app.Spec = appv1.AppSpec{
+			Catalog:   alloyRulesAppCatalog,
+			Name:      alloyRulesChartName,
+			Namespace: alloyRulesDeployInNamespace,
+			Version:   s.AlloyAppVersion.String(),
+			Config: appv1.AppSpecConfig{
```

Comment:
Shouldn't this be a user or extra config instead ?

Reply:
Why? It's the only config the app has

Reply:
I though maybe "something" else would use this field, maybe app-operator ?

Reply:
No it's used by konfigure

Reply:
But that's only for collection apps